File: Neato-SDK/neato-sdk-android/src/main/java/com/neatorobotics/sdk/android/robotservices/RobotService.java
package com.neatorobotics.sdk.android.robotservices;

import androidx.annotation.VisibleForTesting;

import com.neatorobotics.sdk.android.nucleo.NucleoClient;


/**
 * Neato
 * Created by Marco on 04/01/2017.
 * Copyright © 2016 Neato Robotics. All rights reserved.
 */

public abstract class RobotService {
    @VisibleForTesting
    public NucleoClient nucleoClient = new NucleoClient();
}

File: Neato-SDK/neato-sdk-android/src/main/java/com/neatorobotics/sdk/android/robotservices/RobotService.kt
/*
 * Copyright (c) 2019.
 * Neato Robotics Inc.
 */

package com.neatorobotics.sdk.android.robotservices

import com.neatorobotics.sdk.android.NeatoUser
import com.neatorobotics.sdk.android.clients.beehive.Beehive
import com.neatorobotics.sdk.android.clients.beehive.BeehiveRepository
import com.neatorobotics.sdk.android.clients.nucleo.NucleoRepository

abstract class RobotService(
        open var beehiveRepository: BeehiveRepository = BeehiveRepository(Beehive.URL),
        open var nucleoRepository: NucleoRepository = NucleoRepository())
